webapp-django/crashstats/crashstats/forms.py

```diff
@@ -519,3 +519,35 @@ class GraphicsReportForm(BaseForm):
 
     date = forms.DateField()
     product = forms.CharField(required=False)
+
+
+class ExploitabilityReportForm(BaseForm):
+
+    product = forms.CharField()
+    version = forms.CharField(required=False)
+
+    def __init__(self, *args, **kwargs):
+        self.available_products = kwargs.pop('available_products')
+        super(ExploitabilityReportForm, self).__init__(*args, **kwargs)
+
+    def clean_product(self):
+        value = self.cleaned_data['product']
+        if value not in self.available_products:
+            raise forms.ValidationError(
+                '{} is not an available product'.format(value)
+            )
+        return value
```

Comment:
Would that not be better done with a ``ChoiceField``? You can feed it a list of choices at runtime in the ``__init__``. Some forms already do something similar, like here: https://github.com/mozilla/socorro/blob/master/webapp-django/crashstats/supersearch/forms.py#L47-L61